Src/Newtonsoft.Json/Utilities/AsyncUtils.cs

```diff
@@ -92,6 +92,16 @@ public static Task<int> ReadAsync(this TextReader reader, char[] buffer, int ind
             Debug.Assert(reader != null);
             return cancellationToken.IsCancellationRequested ? FromCanceled<int>(cancellationToken) : reader.ReadAsync(buffer, index, count);
         }
+
+        public static bool IsCompletedSucessfully(this Task task)
+        {
+            // IsCompletedSucessfully is the faster method, but only currently exposed on .NET Core 2.0
+#if NETCOREAPP2_0
```

Comment:
I find it unlikely @JamesNK will want to target .NET Core given the availability of .NET Standard. Is `Task.Status` all that slow?

Reply:
Is why I didn't add a target :)

[`Task.Status`](https://github.com/dotnet/coreclr/blob/3758e847988cf8d0e950205657dd0bbdd6818fe6/src/mscorlib/src/System/Threading/Tasks/Task.cs#L1338-L1360) is a pretty ugly constructed property that doesn't inline

Reply:
It was an internal property previously that was exposed post .NET Standard 2.0 https://github.com/dotnet/corefx/issues/16745

Reply:
So the `NETCOREAPP2_0` directive is in case he wants to add it later or for users that source import Json.NET?

Yeah, `Task.Status` does look pretty ugly. I guess an alternative would be `t.IsCompleted && !t.IsFaulted && !t.IsCanceled` which might inline.

Reply:
Yeah, and moving the test to one place (so for example if changed to the 3 tests, instead of `.Status == TaskStatus.RanToCompletion` its only one place that changes - though the 3 tests is pretty horrible too :)

Reply:
I don't plan to add .NET Core 2 target but centralizing testing task completion like this is a good change.